test cases/rust/17 structured sources/meson.build

```diff
@@ -0,0 +1,28 @@
+project('structured input', 'rust')
+
+foo_mod_rs = configure_file(
+  input : 'src/foo.rs.in',
+  output : 'mod.rs',
+  configuration : {'message' : 'Hello, World!'},
+)
+
+conf_file = executable(
+  'main_conf_file',
+  structured_sources(
+    'src/main.rs',
+    {'foo' : [foo_mod_rs]},
```

Comment:
Maybe useful to add a usage of `''` in the test.

Reply:
oh, that's an interesting point. I didn't meant to expose the option of '' in meson DSL. Adding an error for that.

Reply:
Oh whoops. I misunderstood that then :).

Reply:
That's a good catch. I've added an error and a test for said error.